File: palindrome/palindrome.go
package main

import "fmt"

func checkPalindrome(input string) bool {
	for i := 0; i < len(input)/2; i++ {
		if input[i] != input[len(input)-1-i] {
			return false
		}
	}
	return true
}

func main() {
	fmt.Print("Enter the string to check palindrome: ")
	var input string
	fmt.Scanln(&input)

	isPalindrome := checkPalindrome(input)
	if isPalindrome {
		fmt.Println("The entered value is pallindrome")
	} else {
		fmt.Println("The entered value is not pallindrome")
	}
}

File: palindrome/palindrome.ex
defmodule Palindrome do
  @moduledoc """
  Checks whether input string is a palindrome.
  """
  def is_palindrome?(""), do: nil
  def is_palindrome?(str) do
    check_palindrome(str, String.reverse(str))
  end

  defp check_palindrome(str, str), do: true
  defp check_palindrome(_str, _str1), do: false
end
